official/wasm/pkg/goanalysis/goanalysis_test.go

```diff
@@ -0,0 +1,505 @@
+package goanalysis
+
+import (
+	"testing"
+
+	"github.com/ignite/cli/v28/ignite/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppendImports(t *testing.T) {
+	existingContent := `package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	fmt.Println("Hello, world!")
+}`
+
+	type args struct {
+		fileContent      string
+		importStatements []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+		err  error
+	}{
+		{
+			name: "Add single import statement",
+			args: args{
+				fileContent:      existingContent,
+				importStatements: []string{"strings"},
+			},
+			want: `package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+func main() {
+	fmt.Println("Hello, world!")
+}
+`,
```

Comment:
Maybe tests would benefit if `embed` is used in conjunction with `testdata/` folder for the "golden" code snippets

Reply:
I wonder if the golden files will help; we should have more than ten. Maybe we can keep it this way since we will move these functions to the CLI